parsers/basic.c

```diff
@@ -162,7 +190,7 @@ static void extract_dim (char const *pos, BasicKind kind)
 
 	for (; isIdentChar (*pos); pos++)
 		vStringPut (name, *pos);
-	makeSimpleTag (name, kind);
+	makeSimpleTagWithNamespace (name, kind);
```

Comment:
In ctags, we call a name combining with its namespace a qualified name.
Qualified names should be emitted only when `--extras=+q` option is given.
If the option is not provided, ctags should print only non-qualified name.
I'm not good at English. So let's me show an example.

input.c:
```c
struct point {
  float x, y;
};
```
```
$ ./ctags -o - /tmp/input.c
point	/tmp/input.c	/^struct point {$/;"	s	file:
x	/tmp/input.c	/^  float x,y;$/;"	m	struct:point	typeref:typename:float	file:
y	/tmp/input.c	/^  float x,y;$/;"	m	struct:point	typeref:typename:float	file:
$ ./ctags -o - --extras=+q /tmp/input.c
point	/tmp/input.c	/^struct point {$/;"	s	file:
point::x	/tmp/input.c	/^  float x,y;$/;"	m	struct:point	typeref:typename:float	file:
point::y	/tmp/input.c	/^  float x,y;$/;"	m	struct:point	typeref:typename:float	file:
x	/tmp/input.c	/^  float x,y;$/;"	m	struct:point	typeref:typename:float	file:
y	/tmp/input.c	/^  float x,y;$/;"	m	struct:point	typeref:typename:float	file:
```
Here `point::x` and `point::y` are the example of the qualified names.
Your `makeSimpleTagWithNamespace` emits qualified names even when `--extras=+` is not given.
It breaks the consistency of the sematics of the ctags' cmdline.

Instead I would like you to look at the `struct:point` in the example output.
We call it "scope field". The namespace for a name should be stored to the scope field of the name.

Storeing the namespace to the scope field of the name is much important than emitting qualified names because, if the scope field for a name is filled, a client tool reading a tags file can generate qualified names by itself.